tests/test_exceptions.cpp

```diff
@@ -32,6 +32,13 @@ class MyException3 {
 public:
     explicit MyException3(const char * m) : message{m} {}
     virtual const char * what() const noexcept {return message.c_str();}
+    // Rule of 5 BEGIN: to preempt compiler warnings.
+    MyException3(const MyException3&) = default;
+    MyException3(MyException3&& other) = delete;
+    virtual MyException3& operator=(const MyException3& other) = delete;
```

Comment:
Should this be virtual?

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-copy-assignment says "no"

Reply:
Also, why delete copy-assignment, while you can copy-construct?

But as @bstaletic says, it's just tests. So I'm fine with just adding the copy-ctr to make the compiler happy, and forget about everything else. (Same with the comments; this is just good practice, so the comments feel a bit like overkill, personally.)

Reply:
> Should this be virtual?

In the face of ambiguity ... dunno! :-)
I'm using trial-and-error here to see what works across our ~60 CI actions.
Currently I'm experimenting with `default` vs `delete`, aiming to minimize `default`.
Recommendations appreciated, I'll try them out.

Reply:
> https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-copy-assignment says "no"

Thanks!
For completeness:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c63-make-move-assignment-non-virtual-take-the-parameter-by--and-return-by-non-const-